### PR TITLE
Add time limit to query log tool query.

### PR DESCRIPTION
**Pull request by cyulysses-corp**
Allow users to specify a time limit within which the query logs would be retrieved from the QryLogsV table, reducing the retrieval cost for long-standing databases.

### CODE REVIEW

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/subcommand/ExtractSubcommand.java [hunk 1]
```diff
@@ -109,6 +135,22 @@ void setOutputPath(String pathString) {
...
+        "The local start of the time range for the query logs to retrieve.",
+        "The format is 'yyyy-mm-ddThh:mm:ss[.n...]'. Example: '2007-12-03T10:15:30.233333'."
+      })
+  private String startTimeString;
+
+  @Option(
+      names = "--qrylog-timerange-end",
+      description = {
+        "The local end of the time range for the query logs to retrieve.",
+        "The format is 'yyyy-mm-ddThh:mm:ss[.n...]'. Example: '2007-12-03T10:15:30.233333'."
```
Comment (haselbach) [inline src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/subcommand/ExtractSubcommand.java:150]: This is great.  However, could we make this easier for the user to also allow just providing the day, i.e., for a start date we would use the beginning of the day and for the end date, we would use the end of the day if only the day is provided, but not the time?

**Review by haselbach — APPROVED** [review]

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/subcommand/ExtractSubcommand.java (hunk 1 above)
Comment (haselbach) [reply src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/subcommand/ExtractSubcommand.java:150]: Actually, end date should be exclusive. So, end date would also beginning of the day.

Comment (cyulysses-corp) [reply src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/subcommand/ExtractSubcommand.java]: Done.

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/executor/ExtractExecutorImpl.java [hunk 2]
```diff
@@ -76,12 +77,50 @@ public ExtractExecutorImpl(
...
+    // 6-digit max precision to conform with Teradata's TIMESTAMP format.
+    DecimalFormat formatter = new DecimalFormat();
+    formatter.setMinimumIntegerDigits(0);
+    formatter.setMaximumFractionDigits(6);
+    return instantWithoutNanoseconds + formatter.format(instant.getNano() / Math.pow(10, 9));
+  }
+
+  private static void maybeAddTimeRange(
+      SqlScriptVariables.QueryLogsVariables.Builder builder, Arguments arguments) {
+    boolean will_add = false;
```
Comment (haselbach) [inline src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/executor/ExtractExecutorImpl.java:93]: Variable names should use camelCase (i.e., use `willAdd` instead).

Also, the name is not very clear. How about `willAddTimeRange`.

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/executor/ExtractExecutorImpl.java [hunk 3]
```diff
@@ -76,12 +77,50 @@ public ExtractExecutorImpl(
...
+    SqlScriptVariables.QueryLogsVariables.TimeRange.Builder timeRange_builder =
```
Comment (haselbach) [inline src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/executor/ExtractExecutorImpl.java:94]: Please use camelCase for variable names.

**Pull request merged**